### ISSUE TITLE
Placement of newly opened Component Browser dictated by the mouse pointer.

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4265]
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/181076066).
<hr />

## Why
**As** Enso IDE user
**I want** to have nodes placed properly when I've added them using mouse (by dragging out connection or by pressing tab without selection)
**So that** My graph will be nicely laid out

## Notes

Applying the magnet algorithm is not in scope of this task.

## Acceptance Criteria

```gherkin
Scenario: 
GIVEN I have IDE with default newly-created project opened
WHEN I drag out connection from any node and drop it far on the right
OR WHEN I move mouse far on the right and press "tab"
THEN the node searcher is opened under the mouse cursor as described in the design doc.

GIVEN I have IDE with default newly-created project opened
WHEN I drag out connection from the bottom node and drop it under the node
OR WHEN I move mouse cursor under the node and press "tab"
THEN the node searcher is opened with input under the node and aligned on the left

GIVEN I have IDE with default newly-created project opened
WHEN I drag out connection from the top node and drop it between nodes
OR WHEN I move mouse cursor between nodes and nearer upper node and press "tab"
THEN the node searcher is opened with input on the left of the bottom node
```

## Implementation Notes

1. The current `find_free_place_for_node` algorithm should be modified in such a way, that the node seeks a free space _to the left_. (currently, it seeks it _to the right_)
2. When a new node is created (`create_node` function call) we check if the mouse cursor is inside the "previous node dictated placement area" of the source node. Source node is either the node the connection is being dragged from (current algorithm handles that) or a nearest node to the mouse cursor. To determine the nearest node we must iterate over all the nodes in the scene (see `node_areas` in `find_free_place_for_node()` function for an example). 
3. If the cursor is inside this area - we should use the source node position in `find_free_place_for_node` algorithm so that the new node would be placed near the source node.
4. If the cursor is outside this area - we should stick to the current behavior - place a new node under the cursor. 

Pseudo-code:
```
let nearest_node = if cursor_inside_approach_area { Some(nearest_node) } else { None }
let source_node = selected_node OR edge_source OR nearest_node
let position = if let Some(source_node) = source_node { find_free_place_for_node(source_node) } else { cursor_position }
```

[node.rs]: https://github.com/enso-org/enso/blob/14f57271a218d814f602c16be7902b9232e2179c/app/gui/view/graph-editor/src/component/node.rs

The reference image of the "previous node dictated placement area"
![previous node dictated placement area](https://user-images.githubusercontent.com/1623053/149683435-822df787-87fb-4908-903d-1ae8e021c386.png)

## Resources

Old implementation notes (the approach appeared to be overcomplicated): 

1. ~~The [Node component][node.rs] should have a configurable (by code) "previous node dictated placement area" (see attached image below). A new FRP output of type `(bool)` should be added which signals if the mouse cursor is inside this area (see `hover` output implementation - it uses non-trivial approach for performance reasons, to avoid checking each node's coordinates on every frame when testing for hover). Defining the "previous node dictated placement area" should be done in a similar way to how `drag_area` is currently defined for a Node, using the `define_shape_system!` macro (this will automatically provide events like `mouse_over` and `mouse_out` for the area).~~
2. ~~Graph Editor should handle this output and react on a new node creation in this area by using `find_free_place_for_node` algorithm. If the mouse cursor is outside of this area, the node should be created normally under the cursor (i.e. using the same algorithm as used before this task). The place in code where the position is settled is in `create_node` method. If the mouse cursor is inside of this area of some node, this node's ID should be used by `create_node` for calculating the node searcher position.~~

## Blockers:
[#180873545](https://www.pivotaltracker.com/story/show/#180873545) resolved

## Comments:
**Technical Design** review set to **pass** (Adam Obuchowicz - Feb 17, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-02-17):

**Today:** Merged #180459079. Started reading #181076066 but stumbled upon problems with WebGL-related panics, reported as #181296538. Spent rest of the day diagnosing #181296538 (trying older commits, trying clean rebuilds, trying restarts and closing other applications running in parallel), with major help from MM first and especially from WD later. It should be finished by 2022-02-23.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Continue investigating #181296538: upgrade to Chrome Beta; if not helping, try downgrading to older Chrome; if not successful, try running `./run watch` with Electron as browser. Once a workaround is found, switch back to #181076066. (Enso Bot - Feb 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-02-17):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

It seems that a workaround of using Chrome Beta solves the problem for me, so no further delay is expected.

**Delay Cause:** IDE panics due to lost WebGL context when opening `./run watch` IDE with Chrome. Reported #181296538 and spent most of the day at diagnosing the circumstances. (Enso Bot - Feb 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-18):

**Today:** For #181296538, tested with Chrome Beta, which was successful. Added a note with a link to the discord thread in the ticket, then closed the ticket as resolved (Accepted). Started working on #181076066; read the description; had a call with IB to clarify some questions; added the clarifications to the ticket description; started defining the shape of the area described in the ticket (working name: "approach area") and connecting its events. It should be finished by 2022-02-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Position the approach area correctly (should not be centered on the node, but below it). In Graph Editor, track the "approached nodes" (i.e. node[s] for which the approach area is hovered by mouse); display debugging information when approach areas are hovered/unhovered, and which nodes are tracked as approached in Graph Editor. Later: use the "approached node[s]" information in `create_node` function, and verify that the logic works correctly. (Enso Bot - Feb 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-21):

**Today:** Positioned & sized the "approach area" correctly (so that it's not centered on the node, but below it). Passed the "approach area" hover status to the `create_node` function. Changed `create_node` to position the node correctly when TAB is pressed while "approach area" is hovered. It should be finished by 2022-02-24.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Find fixes for some bad logic: "approach area" hover is lost when output node is hovered; clicking on "approach area" while dragging out a connection does not create a node. (In other words, this would seem to boil down to a question: how to make the "approach area" (and maybe other areas too) transparent/passthrough to clicks?) - consider asking IB/AO. (Enso Bot - Feb 21, 2022)
<hr />
**Technical Design** review set to **revise**

The approach appeared to be overcomplicated (Ilya Bogdanov - Feb 22, 2022)
<hr />
**Technical Design** review set to **pass** (Ilya Bogdanov - Feb 22, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-02-22):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

**Delay Cause:** initial Technical Design of the task led to difficult to resolve issues, which after discussion led to discovery that the design was overbroad, and the EPIC Design Doc only asks for a simpler solution, which can be implemented easier, but needs to be redone from scratch. (Enso Bot - Feb 22, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-22):

**Today:** Started a discussion with IB and AO asking how to track mouse events for overlapping Views. This ended up leading to a change in Technical Design of the task (reduction of scope and simplification). Based on the redesign, in `create_node()`, added a loop over all nodes to find the nearest one to the mouse pointer; also added a calculation to check if mouse pointer is in the "approach area" of a node; added some initial debugging messages. It should be finished by 2022-02-25.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Debug why the initial draft of finding the nearest node does not work as I would expect - whether its logic is buggy or it needs to be changed to a less simplistic method. Extend `create_node()` so that Component Browser placement takes into account the nearest node approach area calculations. (Enso Bot - Feb 22, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-23):

**Today:** Debugged and improved the handling of "placement area" in `create_node()`. Added handling of the "dropped edge" case. Improved the UX of "nearest node" logic by changing the underlying algorithm, from treating a node as a point for calculation of distance, into treating a node as a line segment - this makes behavior much predictable and intuitive from UX perspective. Refactored the code from an initial prototype to more readable style. Also, attended the company AMA meeting. It should be finished by 2022-02-25.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Test the feature in a `./run dist` build. Record a screencast. Review the code, possibly adding documentation, and open a PR if enough time. Talk with Team Lead how to address the issue of connections (edges) dropped on `type_label` not leading to node creation. (Enso Bot - Feb 23, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-24):

**Today:** Recorded a screencast in `./run dist` build. Polished the code for a PR and opened a PR. Reported an issue noticed during development of the task (#181369980). Started a Task Review of #180916527. Got a PR code-review of #181076066, had a quick call with the reviewer, got some clarifications, suggestions, and guidance. It should be finished by 2022-02-25.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Address code-review comments in #181076066. Continue Task Review of #180916527. If time permits, start Technical Design of #181076145, or Implementation of #180887079 (discuss the choice with EPIC Leader). (Enso Bot - Feb 24, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-25):

**Today:** Addressed code-review comments in #181076066 and re-requested review. Switched back to Acceptance/QA Review of #180916527: compiled a development version of the Engine; started an IDE instance connected to the local Engine (to get it to initialize with sample project's contents); got some guidance from DB on how to add a local Enso library and on some subtle aspects of formatting Engine API messages; did initial round of Acceptance/QA testing by sending raw API messages to Engine's websocket; collected a list of discrepancies found between design doc, feature behavior, and API docs, then posted them to a place agreed with DB (https://github.com/enso-org/enso/issues/3304), and linked it from a comment in #180916527. Had a "virtual coffe" meeting with IB. It should be finished by 2022-02-25.

**Next Day:** Next day I will be working on the [##180916527 task](https://www.pivotaltracker.com/story/show/180916527). Continue Acceptance/QA Review of Task #180916527. Then, start Technical Design of #181076145, or Implementation of #180887079 (discuss the choice with EPIC Leader). (Enso Bot - Feb 25, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-02-28):

**Summary:** There is **3 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

Uncertain as to whether changes will be accepted; I started requesting a call with WD and AO in case further review comments would be added.

**Delay Cause:** Addressing review comments, including requests to change function names that were already consulted with AO;  finding good names seems challenging. Also, missed adding function comments in the original PR, due to confusion about code standards in the file, including very recent changes by other contributors.

**Possible solutions:** Have a call with WD and AO, to try and quickly clarify what specific changes could be done to the PR to make it mergeable. (Enso Bot - Feb 28, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-02-28):

**Today:** Addressed code-review comments. Switched to #180916527 and completed the Acceptance/QA Review of the Task. Started working on Technical Design of #181076145. It should be finished by 2022-02-28.

**Next Day:** Next day I will be working on the [##180916527 task](https://www.pivotaltracker.com/story/show/180916527). Send #181076066 to Task Review if PR gets Approved, or try to schedule a meeting with WD and AO to discuss how to resolve it. Continue working on the Technical Design of #181076145. Estimate and start working on implementation of #180887079. (Enso Bot - Feb 28, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-03-01):

**Summary:** There is **2 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

**Delay Cause:** (1) Got a clarification in response to a question about unspecified aspect of design, but it was different than anticipated, and requires changes in distance calculation code; lesson learned: should have split off the question from code review process and should have posted it on Discord earlier; (2) my progress in learning how to name things in a way that would be considered clear and easy to understand by reviewers appears to be slow. (Enso Bot - Mar 1, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-01):

**Today:** Had 4 meetings. Discussed with WD and AO code changes to the PR during an interactive code review call. Started implementing suggestions from the call. It should be finished by 2022-03-02.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Continue updating #181076066 based on code review suggestions (including a change to calculations of the distance between a node and a mouse pointer, to take into account visualization area). If time permits, do 2nd round of Acceptance/QA Review on #180916527 (based on fixes in PR 3308). (Enso Bot - Mar 1, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-03-02):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

**Delay Cause:** (1) Including the visualization in node's bounding box shows up to be less trivial than anticipated (had a setback due to misunderstanding how visualization is positioned relative to node; visibility of visualization is not trivial and needs further analysis, including proper handling of error visualization visibility; some time spent on trying to come up with correct names). (2) Also, spent 1 hour doing round 2 of Acceptance/QA Review for #180916527, to unblock DB on it. (Enso Bot - Mar 2, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-02):

**Today:** Did a 2nd round of Acceptance/QA Review on #180916527, and reported a few discrepancies found. Switched back to #181076066. Implemented calculation of visualization's bounding box and adding it to node's bounding box if visualization is enabled; modified the algorithm for finding a node closest to the mouse pointer so that it now uses bounding boxes of nodes. Started modifying "alignment area" calculations to take into account visualization. It should be finished by 2022-03-03.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Continue working on #181076066: finish "alignment area" calculations, do testing, apply remaining code-review comments, do another self-review of func & var names, re-request code-review. (Enso Bot - Mar 2, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-03-03):

**Today:** Continued work on including visualization in bounding box of a node. Wrote tests. Applied code-review comments. Re-requested PR review. It should be finished by 2022-03-03.

**Next Day:** Next day I will be working on the [##181076145 task](https://www.pivotaltracker.com/story/show/181076145). Apply any further code-review comments to #181076066 if present. Switch to Technical Design for #181076145. (Enso Bot - Mar 4, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-03-04):

**Summary:** There is **4 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

(4 days = 2 workdays + weekend). Had a call with Sylwia today, we discussed the situation with my PRs being slow to merge, and some thoughts and ideas how to try and improve it.

**Delay Cause:** Got numerous further comments in code review, that I did not manage to address all today, due to required rethinking of various parts of the code and/or further exploring/learning the codebase. (Enso Bot - Mar 4, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-04):

**Today:** Worked on addressing new code-review comments. Had a call with Sylwia today, we discussed the situation with my PRs taking long time in reviews, and some thoughts and ideas how to try and improve it. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Continue addressing code-review comments on #181076066. Re-request review. Switch to Technical Design for #181076145. (Enso Bot - Mar 4, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-07):

**Today:** Finished addressing recent code-review comments. Re-requested review from WD. Checked and Approved PR 3319. Updated, checked, and re-requested review for PR 3313. It should be finished by 2022-03-07.

**Next Day:** Next day I will be working on the [##181076145 task](https://www.pivotaltracker.com/story/show/181076145). Apply any further code-review comments to #181076066 if present. Switch to Technical Design for #181076145. (Enso Bot - Mar 7, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-03-09):

**Summary:** There is **2 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

(1 day spent on other task, 1 day on this one.) Merging the task is now frozen pending a fix for PR 3328. The PR is Approved, so no other delays are currently expected unless new code review comments show up, or Task Review asks for some changes.

**Delay Cause:** Yesterday I was working on a different task (Tech Design). Today I got new code review comments for #181076066 and addressed them. (Enso Bot - Mar 9, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-09):

**Today:** Addressed comments from a new code-review. Also, modified the Implementation Notes of #181076145 to include suggestions from the review of its Technical Design. It should be finished by 2022-03-09.

**Next Day:** Next day I will be working on the [##181076145 task](https://www.pivotaltracker.com/story/show/181076145). Start implementation work on the new Task. (Enso Bot - Mar 9, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-03-10):

**Summary:** There is **2 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this sprint.

I am not certain about the delay size: it seems challenging for me to get my PRs to a shape that would get accepted for merging.

**Delay Cause:** Got numerous new comments in code-review and need to address them.

**Possible solutions:** (1) If more of the code-review comments/suggestions from later code-reviews were provided already in earlier reviews, I would be able to address them earlier. (2) If more of the code-review comments/suggestions were more specific and provided concrete examples, instead of being open-ended, I'd have more examples to learn from. Unfortunately, the existing codebase appears to not be a good source of learning/inspiration/examples of good practices and style: when I copy fragments of code or text from the existing codebase into my PRs, I typically receive comments calling them out as not good and needing to be fixed. (Enso Bot - Mar 11, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-03-10):

**Today:** Started addressing comments from a new code-review by WD. Carried out discussions with the reviewers to get some clarifications for aspects that weren't clear to me. It should be finished by 2022-03-11.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/181076066). Continue addressing code-review comments, re-request a code-review. Switch to #181076145. (Enso Bot - Mar 11, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for the provided date (2022-03-11):

**Today:** Finished addressing comments from the most recent code-review. Re-requested a code review. It should be finished by 2022-03-11.

**Next Day:** Next day I will be working on the [##181076145 task](https://www.pivotaltracker.com/story/show/181076145). Apply any further code-review comments to #181076066 if present. Switch to implementation work on #181076145. (Enso Bot - Mar 16, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-03-17):

**Summary:** There is **7 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Mateusz sick days, my paternity time off and probably the old estimate did not include Acceptance/QA phase. (Enso Bot - Mar 18, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-03-17):

**Progress:** Did the manual testing for mouse-oriented node placement. Also wrote the automatic ones, but there is one strange, non-deterministic regression in integration tests. The rest of time I spent catching up with things after time off. It should be finished by 2022-03-18. (Enso Bot - Mar 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for the provided date (2022-03-16):

**Progress:** Addressed a few post-Approval code-review comments from WD. It should be finished by 2022-03-18. (Enso Bot - Mar 18, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-03-17):

**Progress:** Addressed new code-review comments from AO. It should be finished by 2022-03-18. (Enso Bot - Mar 18, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for the last Friday (2022-03-18):

**Summary:** There is **3 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** I fought with build script issue instead of finishing tests (Enso Bot - Mar 21, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for the last Friday (2022-03-18):

**Progress:** Tried to track down the issues with tests, but spotted other one instead, in our build scripts. Debugging took a lot of time, but finally found the core issue. It should be finished by 2022-03-21. (Enso Bot - Mar 21, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-21):

**Progress:** Merged new `develop` into the PR's branch, resolving conflicts. Test the merged code to verify that the functionality works correctly. It should be finished by 2022-03-21. (Enso Bot - Mar 21, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-03-21):

**Summary:** There is **3 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 0 days delay** for the delivery of this weekly plan.

Giving more time, as I also want to finish my old PR

**Delay Cause:** Still no luck in investigating integration tests issue. (Enso Bot - Mar 22, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for yesterday (2022-03-21):

**Progress:** Added many debug logs. It seems that nan-s are returned from screen_to_object_space method. But adding debug logs there does not display logs in console for some reason + it makes the issue almost not reproducible It should be finished by 2022-03-24. (Enso Bot - Mar 22, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-03-22):

**Progress:** With Michał's help we get more info from bug and have some guesses what's going on. It should be finished by 2022-03-24.

**Next Day:** Yet another day of hope for fixing the issue. (Enso Bot - Mar 22, 2022)
<hr />
**Adam Obuchowicz** reports a new **🔴 DELAY** for yesterday (2022-03-28):

**Summary:** There is **6 days** delay in implementation of the [Placement of newly opened Component Browser dictated by the mouse pointer. (#181076066) task](https://www.pivotaltracker.com/story/show/181076066).
It **will cause 6 days delay** for the delivery of this weekly plan.

Today decided to leave the bug and fix it in a separate issue/PR

**Delay Cause:** Sick leave + one more days of investigating bug. (Enso Bot - Mar 29, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-03-29):

**Progress:** Decided to merge the PR ASAP, only one tiny thing remains in automatic tests. Many meetings today, also getting things working on MacOS. It should be finished by 2022-03-30.

**Next Day:** Write one more thing, merge PR and finally unblock Ilya. (Enso Bot - Mar 29, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for today (2022-03-30):

**Progress:** Extended tests a little bit, fix one stupid mistake. Almost merged PR, however at some point I could not run integration tests for some weird reason It should be finished by 2022-03-30.

**Next Day:** Merging PR (Wojciech already said me how to fix the issue). After that writing required techdocs, and move to designing the issue. (Enso Bot - Mar 30, 2022)
<hr />
**Acceptance/QA** review set to **pass** (Adam Obuchowicz - Mar 31, 2022)
<hr />
